### PR TITLE
[receiver/gitproviderreceiver] rename to githubreceiver

### PR DESCRIPTION
**Pull request by adrielp**
**Description:** <Describe what has changed.>
Renames `gitproviderreceiver` to `githubreceiver` to better align with OpenTelemetry Semantic Conventions v1.27.0+ and allow for tracing and log signals in the future.

### CODE REVIEW

File: .chloggen/gitprovider-rename.yaml
Comment (andrzej-stencel) [inline .chloggen/gitprovider-rename.yaml:7]:
```suggestion
component: gitproviderreceiver
```

**Review by andrzej-stencel — APPROVED** [review]
After the rename, the "provider" part in the name seems a bit (more) awkward. (I think it was me who originally came up with this :zany_face:). I'm not sure what other name would be better.

- VCS receiver?
- VCS Metrics receiver?
- VCS Repository receiver?
- Repository Change Metrics receiver? :sweat_smile: 
- ...

**Comment by adrielp** [comment]
> After the rename, the "provider" part in the name seems a bit (more) awkward. (I think it was me who originally came up with this 🤪). I'm not sure what other name would be better.
> 
> * VCS receiver?
> * VCS Metrics receiver?
> * VCS Repository receiver?
> * Repository Change Metrics receiver? 😅
> * ...

I had the same thought @andrzej-stencel. Technically it's still whatever a VCS provides, however, I think `vcsmetricsreceiver` might be a better option since it's focused on metrics only from version control systems.

File: receiver/vcsmetricsreceiver/README.md
Comment (crobert-1) [inline receiver/vcsmetricsreceiver/README.md:39]: This README has a link below to the Liatrio distribution, just a couple comments:

1. I believe the referred to distribution is `downstream` from `contrib`, as it relies on `contrib`, not vice versa.
2. My actual comment, the referenced config file will need to be updated to account for this name change.

In general I'd prefer to have configuration examples in this repository to avoid stale information. It's a pretty common issue to forget to update READMEs with updated configuration information, and it will only be harder to keep all examples and information up to date if there are more places to keep updated. I'd appreciate input though, maybe there's a good reason to keep location of this information the way it is.

**Review by crobert-1 — COMMENTED** [review]
Out of ignorance, is there any potential of this receiver being able to support logs or traces in the future? (If so, we may not want to have `metrics` in the name.)

File: receiver/vcsmetricsreceiver/README.md
Comment (adrielp) [reply receiver/vcsmetricsreceiver/README.md:39]: @crobert-1  - This pr is just a move, but I'm happy to update the docs minimally where needed in this PR. 

The mention of the Liatrio distro is there because the component has largely been developed there first, aside for this rename. Being that, there are features in that distro (like a GitLab scraper, and CVE Metrics from GitHub) that haven't been contributed here yet. That being said, that one sentence can be removed if necessary. IIRC (it's been almost a year now) this was added intentionally. Especially for those that want to use the component and don't build their own distros since this hasn't been built into contrib yet. Of course, this was a different time because back then these types of distros were supported as being linked directly as a distro in the generated docs. 

The config examples in the README are up-to-date. Happy to remove that mention though. I would suggest eventually adding another repository we have that runs all this locally through Kubernetes for those that want to get running locally quickly too. 

I did say "largely" as of late I've been doing most of these contributions on my own time. These VCS updates haven't made it into the Liatrio distro as of yet. I could go either way, so if it's important to remove I can align.

**Comment by adrielp** [comment]
> Out of ignorance, is there any potential of this receiver being able to support logs or traces in the future? (If so, we may not want to have `metrics` in the name.)

That is a good question. I never planned for that. There is a [GitHub Action Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27460) that I'm getting ready to contribute that will receiver events from GitHub, turn them into traces, and additional receive the workflow logs. In that structure, it's essentially a web hook receiver. 

Is there precedence where both scraping & web hook style receiving occur within the same receiver component? -- If so, I could see that tracing component being combined with this one?

**Review by andrzej-stencel — APPROVED** [review]

File: receiver/vcsmetricsreceiver/README.md
Comment (crobert-1) [reply receiver/vcsmetricsreceiver/README.md:39]: Thanks for providing the extra context, I wasn't familiar with the history here 👍 In that case, it's okay with me to leave it as-is, the reference makes sense in this case. As long as we can keep the references up to date to avoid conflicting information it's fine with me.

**Review by crobert-1 — APPROVED** [review]
I'm aware that more discussion is going on in other issues and CNCF Slack around direction to go here, but I'm approving as I'm fine either way. We can move this PR forward once there's a resolution there. (If I'm understanding properly).

**Comment by adrielp** [comment]
> I'm aware that more discussion is going on in other issues and CNCF Slack around direction to go here, but I'm approving as I'm fine either way. We can move this PR forward once there's a resolution there. (If I'm understanding properly).

sounds good, appreciate it @crobert-1 🙇🏼

**Comment by adrielp** [comment]
Cross posting the outcome from the SIG decision today https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27460#issuecomment-2302537838

will update this PR to reflect.

**Review by mwear — APPROVED** [review]

**Comment by adrielp** [comment]
This is now ready for merge. In future iterations I may make some small internal shifts given that we will be allowing for more signals within this singular component.

**Review by crobert-1 — APPROVED** [review]
Thanks for making the rename! One potential change here would be to move the `internal/scraper/githubscraper/` directory up a level to just be `internal/scraper`, since there's no need to qualify which scraper is being used. Just a thought though, not needed as a part of the rename.

**Comment by crobert-1** [comment]
@andrzej-stencel I know you've already approved, but would you be up for reviewing again with this new rename? Just want to make sure we're agreed here. 👍

**Review by andrzej-stencel — APPROVED** [review]

**Comment by andrzej-stencel** [comment]
> @andrzej-stencel I know you've already approved, but would you be up for reviewing again with this new rename? Just want to make sure we're agreed here. 👍

Sure! :rocket:

**Comment by adrielp** [comment]
> Thanks for making the rename! One potential change here would be to move the `internal/scraper/githubscraper/` directory up a level to just be `internal/scraper`, since there's no need to qualify which scraper is being used. Just a thought though, not needed as a part of the rename.

I agree! that's the next thing to do outside this PR!

**Comment by mx-psi** [comment]
I think we can merge this once conflicts are resolved

**Comment by adrielp** [comment]
@mx-psi - conflicts resolved. Unrelated and unexpected failure observed in build & test workflows for the redaction processor. Fixed once #34937 is merged. All workflows are good otherwise.

**Comment by mx-psi** [comment]
Looks like this

```
  [✖] https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/githubreceiver/internal/scraper/githubscraper/README.md#github-limitations → Status: 404
```

Is a real error though?

**Comment by adrielp** [comment]
> Looks like this
> 
> 
> 
> ```
> 
>   [✖] https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/githubreceiver/internal/scraper/githubscraper/README.md#github-limitations → Status: 404
> 
> ```
> 
> 
> 
> Is a real error though?

@mx-psi it will be resolved once merged into main because this is going to a main ref for the documentation which doesn't currently exist given that this is a component rename

File: receiver/githubreceiver/README.md
Comment (mx-psi) [inline receiver/githubreceiver/README.md:93]: I think something like this would also fix it and it (arguably) makes more sense to do this for links internal to this repository
```suggestion
[ghsread]: internal/scraper/githubscraper/README.md#github-limitations
```

**Pull request merged**
